crates/sui-adapter/src/adapter.rs

```diff
@@ -561,13 +564,39 @@ fn handle_transfer<
             //   2. Created in this transaction (in `newly_generated_ids`)
             //   3. Unwrapped in this transaction
             // The following condition checks if this object was unwrapped in this transaction.
-            if old_object.is_none() && !newly_generated_ids.contains(&obj_id) {
-                // When an object was wrapped at version `v`, we added an record into `parent_sync`
-                // with version `v+1` along with OBJECT_DIGEST_WRAPPED. Now when the object is unwrapped,
-                // it will also have version `v+1`, leading to a violation of the invariant that any
-                // object_id and version pair must be unique. Hence for any object that's just unwrapped,
-                // we force incrementing its version number again to make it `v+2` before writing to the store.
-                move_obj.increment_version();
+            if old_object.is_none() {
+                // Newly created object
+                if newly_generated_ids.contains(&obj_id) {
+                    // TODO: log NewObject event
+                } else {
+                    // When an object was wrapped at version `v`, we added an record into `parent_sync`
+                    // with version `v+1` along with OBJECT_DIGEST_WRAPPED. Now when the object is unwrapped,
+                    // it will also have version `v+1`, leading to a violation of the invariant that any
+                    // object_id and version pair must be unique. Hence for any object that's just unwrapped,
+                    // we force incrementing its version number again to make it `v+2` before writing to the store.
+                    move_obj.increment_version();
+                }
+            } else if let Some((_, old_obj_ver)) = old_object {
+                // Some kind of transfer since there's an old object
+                // Add an event for the transfer
+
+                match recipient {
+                    Owner::AddressOwner(addr) => state_view.log_event(SuiEvent::TransferObject {
+                        object_id: obj_id,
+                        version: old_obj_ver,
+                        destination_addr: addr,
+                        type_: TransferType::ToAddress,
+                    }),
+                    Owner::ObjectOwner(new_owner) => {
+                        state_view.log_event(SuiEvent::TransferObject {
+                            object_id: obj_id,
+                            version: old_obj_ver,
+                            destination_addr: new_owner,
+                            type_: TransferType::ToObject,
+                        })
+                    }
+                    _ => {}
```

Comment:
if `old_object` is None, does it imply an object creation?

Reply:
Yes.  I might add an event for that as well.